rfcs/0002-assetid.md

```diff
@@ -0,0 +1,107 @@
+
+Note that this is the first part in a series of RFCs I intend to write that details the technical aspects of my earlier [Asset Pipeline](https://github.com/amethyst/amethyst/issues/875) proposal.
+
+# Motivation
+[motivation]: #motivation
+
+To realize the vision of the Asset Pipeline it will be important to be able to address loadable assets in a more powerful way than with a filesystem path.
+Container formats such as GLTF and FBX can benefit from AssetID where an ID can be generated for each loadable asset within the file as opposed to referencing the entire source file.
+It is also essential that cross-file references use UUID for Amethyst to handle renaming or moving of files without breaking these references.
+
+The primary benefit of AssetID is to make it easier to write tools for Amethyst that handle assets. With a unified system for identifying assets, we can build a unified system for loading assets or metadata about assets which is super useful when creating editors and visualisers.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+AssetID is an identifier for loadable assets. It is a name and enum for any of three variants.
+AssetUUID - Globally/universally unique identifier for an asset - v4 UUID. Generated by the Asset Pipeline when an asset is imported. Primary way to reference assets for tools and formats that are not edited by humans.
+FilePath - The current way of referencing assets. It will still be supported in scenarios where it is possible.
+URI - A way for users to plug in their own custom asset referencing scheme along with their own resolving systems.
+
+I expect users that write code to primarily reference assets using the FilePath variant. References in data, for example those that are created by a Prefab editor when referencing meshes or textures would use UUID references. This is closer to the expectation of the user since the user did not specify a specific path, but a reference to an asset. 
+
+If a user created a prefab using a prefab editor and references an asset in a file that was later renamed or moved, it would be unexpected for the reference to no longer resolve since the user did not specify a path in the prefab editor. In the case of loading an asset by file path in code, it would be be expected behaviour for the load to fail if a file is removed however.
+
+URI is a general way of referencing an asset that does not exist in the filesystem and was not imported by the asset pipeline. An example of this could be an asset stored in a database somewhere that is specific to the game it is resolved in. URIs are included in AssetID as a point of further extension for game developers and will not be used by any of the initial Asset Pipeline systems.
+The intention is to provide traits that can be implemented to make it easier to integrate custom IDs into the asset loading process.
```

Comment:
This custom trait (let's name it `AsessetAlias`) is redundant with everything except the raw UUID.